modules/rest-util/src/blaze/middleware/fhir/output.clj

```diff
@@ -53,6 +66,14 @@
   (cond-> response body (-> (update :body generate-xml)
                             (ring/content-type content-type))))
 
+(defn- encode-response-binary [{:keys [body] :as response}]
+  (let [content-type (or (-> response :body :contentType fhir-type/value)
+                         "application/octet-stream")]
+    (cond->
+        (ring/content-type response content-type)
+      body (-> (update :body generate-binary)))))
```

Comment:
In case the body is nil, I would also not set the content-type to `application/octet-stream`. it has no value to do it for an empty content. So you can go for the same style as in `encode-response-json` here. Please also write a separate function that takes the body and returns the content-type named `binary-content-type`.